.github/workflows/python_cdk_tests.yml

```diff
@@ -0,0 +1,157 @@
+# THIS WORKFLOW SHOULD BE REPLACED BY A CLEANER ONE ONCE THE PYTHON CDK TESTS CAN BE RUN WITH POETRY
+name: Python CDK Tests
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  S3_BUILD_CACHE_ACCESS_KEY_ID: ${{ secrets.SELF_RUNNER_AWS_ACCESS_KEY_ID }}
+  S3_BUILD_CACHE_SECRET_KEY: ${{ secrets.SELF_RUNNER_AWS_SECRET_ACCESS_KEY }}
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      python_cdk: ${{ steps.changes.outputs.python_cdk }}
+    steps:
+      - name: Checkout Airbyte
+        if: github.event_name != 'pull_request'
+        uses: actions/checkout@v3
+      - id: changes
+        uses: dorny/paths-filter@v2
+        with:
+          # Note: expressions within a filter are OR'ed
+          filters: |
+            python_cdk:
+              - 'airbyte-cdk/python/**/*'
+  run-python-cdk-check:
+    needs:
+      - changes
+    if: needs.changes.outputs.python_cdk == 'true'
+    runs-on: connector-test-large
+    name: Python CDK Tests
+    timeout-minutes: 30
+    steps:
+      - name: Checkout Airbyte
+        uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: "zulu"
+          java-version: "21"
+      - name: Install Pip
+        run: curl -fsSL https://bootstrap.pypa.io/get-pip.py | python3
+      - name: Install Pyenv
+        run: python3 -m pip install virtualenv --user
+      - name: Docker login
+        # Some tests use testcontainers which pull images from DockerHub.
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: cd airbyte-cdk/python
+        run: cd airbyte-cdk/python
+      - name: Run Gradle Check
```

Comment:
```suggestion
      - name: Run Gradle Check on Python CDK
```

Reply:
You need to name this something else in order to make this a required branch check.